SalesforceCommerceCloud/pwa-kit
# Add wrappers for passwordless login helpers in `commerce-sdk-react`

# Description
Implementing wrappers for the passwordless login helpers in `commerce-sdk-isomorphic`. The `isomorphic` changes have not been released yet, so we are merging these changes into the feature branch for now.

How we will configure passwordless login:
```
module.exports = {
    app: {
        login: {
            social: {
                enabled: true,
                idps: ['Google', 'Apple']
            }
            passwordless: {
                enabled: false
                callbackURI: 'callbackURI'
            }
        }
    }
}
```
- Doc link [here](https://salesforce.quip.com/bEb4AhHZtqDy#temp:C:PYX3957020842c542d98fd7a3035)


# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Add wrappers for the following 2 isomorphic helpers:
    - `authorizePasswordless`
    - `getPasswordLessAccessToken`
- Read passwordless login config from the retail react app config file and add `callbackURI` to the Auth class
    - If `callbackURI` is set, then mode is `callback`
        - URI can be absolute or relative !
    - If it is missing yet passwordless login is enabled, and mode is `sms`, then mode param is set to `sms`

# How to Test-Drive This PR
- Go to https://wasatch-mrt-passwordless-test.mrt-storefront-staging.com/login
- Click **Continue Securely** to make the 1st API call, [authorizePasswordlessCustomer](https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-login?meta=authorizePasswordlessCustomer)
  - In the **Network** tab, verify that the `passwordless/login` request is successful with a **200** response status
- Then navigate to https://webhook.site/993f131f-117b-4b4d-a525-72cf895fb32a and grab the `token` from the response body
- In the same login page, input this token where it says **Enter text** then click **Submit** to make the 2nd API call, [getPasswordLessAccessToken](https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-login?meta=getPasswordLessAccessToken)
- Verify that you are successfully logged into the secure account page

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
